tbls/herumi.go

```diff
@@ -36,6 +39,27 @@ func init() {
 // Herumi is an Implementation with Herumi-specific inner logic.
 type Herumi struct{}
 
+// GenerateInsecureKey generates a key that is not cryptographically secure using the
+// provided random number generator. This is useful for testing.
+func (Herumi) GenerateInsecureKey(t *testing.T, random io.Reader) (PrivateKey, error) {
+	t.Helper()
+	for i := 0; i < 100; i++ {
+		b := make([]byte, 32)
+		_, err := random.Read(b)
+		require.NoError(t, err)
+
+		p := new(bls.SecretKey)
+		err = p.Deserialize(b)
+		if err != nil {
```

Comment:
not all random values are valid, seems around 50% are valid.

Reply:
AFAIK `bls.SetRandFunc` can be used to tell the Herumi library to use a specific source of entropy.

Might help with this issue?